### PR TITLE
Fix part01 step15 test

### LINKED ISSUE #897: Fix Part01Step15 Test
Break up the test of Part01Step15 so that each failure is clearly delineated.

### PR DESCRIPTION
**Pull request by mmschaefer**
Resolves #897

### CODE REVIEW

File: src-test/org/etools/j1939_84/controllers/part01/Part01Step15ControllerTest.java [hunk 1]
```diff
@@ -130,8 +131,7 @@ public void testEmptyPacketFailure() {
     }
 
     @Test
-    @Ignore("This test needs broken up")
-    public void testFailures() {
+    public void testActiveDtcFailure() {
         DM1ActiveDTCsPacket packet1 = new DM1ActiveDTCsPacket(
```
Comment (gumbelmj) [inline src-test/org/etools/j1939_84/controllers/part01/Part01Step15ControllerTest.java]: Let's bring this test up to date and use the new pattern of
- One test for Happy Path / No Failures
- One test per failure/warning expected

Additionally, let's remove the use of creating packets with data and instead use the static `create()` methods so it's easier to see what data is being used in the test.

**Review by gumbelmj — CHANGES_REQUESTED** [review]

**Comment by mmschaefer** [comment]
image

New look for step cover test doc reporting

**Review by gumbelmj — APPROVED** [review]

File: src-test/org/etools/j1939_84/controllers/part01/Part01Step15ControllerTest.java [hunk 2]
```diff
@@ -302,44 +321,24 @@ public void testGetStepNumber() {
     }
 
     @Test
+    @TestDoc(value = @TestItem(verifies = "6.1.15.1", description = "Verifies that there is a single 6.1.15 step"))
     public void testGetTotalSteps() {
         assertEquals("Total Steps", 0, instance.getTotalSteps());
```
Comment (battjt) [inline src-test/org/etools/j1939_84/controllers/part01/Part01Step15ControllerTest.java]: "Verifies that there is a single 6.1.15 step", but then is verifies that there are zero steps.

Comment (mmschaefer) [reply src-test/org/etools/j1939_84/controllers/part01/Part01Step15ControllerTest.java]: Removed - should not have been added.  Number of steps represents the times the progress bar will be updated.  They don't care.

**Review by battjt — APPROVED** [review]

**Pull request merged**